### ISSUE TITLE
update queries: EXPR not working

### ISSUE THREAD

**Issue by dbu** [phpcr/phpcr-shell#158]
i tried to run this query:

```
UPDATE [nt:unstructured] AS a INNER JOIN [nt:unstructured] AS t ON ISCHILDNODE(t, a) 
SET EXPR('row.getNode("t").setProperty("originalLinkCaption", row.getNode("a").getPropertyValue("originalLinkCaption")') 
WHERE a.[phpcr:class] = 'Liipch\MagazineBundle\Document\Article' ;
```

but get the exception 
```
[PHPCR\ValueFormatException] Invalid value for property ")": Object values must implement PHPCR\NodeInterface, PHPCR\PropertyInterface or be \DateTime, supplied argument is of class: PHPCR\Shell\Query\ColumnOperand
```

am i doing something stupid?

**Comment by dantleech** [comment]
I don't think so.. looks like a parser issue. Maybe try escaping the quotations?

**Comment by dantleech** [comment]
hmm .. i get the same issue

**Comment by dantleech** [comment]
Try:

````sql
UPDATE [nt:unstructured] AS a INNER JOIN [nt:unstructured] AS t ON ISCHILDNODE(t, a) 
SET t.originalLinkCaption = EXPR('row.getNode("a").getPropertyValue("originalLinkCaption")') 
WHERE a.[phpcr:class] = 'Liipch\MagazineBundle\Document\Article' ;
````

**Comment by dbu** [comment]
i tried that, it leads to the bug described in #160

**Comment by dantleech** [comment]
Hmm, it works for me locally (it parses the query and updates 0 rows). Are you using the latest version? (beta-3)

**Comment by dbu** [comment]
oh, actually not. i will try that out and report back.

**Comment by dbu** [comment]
ok, your proposal works. but the thing i reported still gives me the same on beta3 and i don't think i see a typo in there. so maybe a bug in the expression parser?

functions got case sensitive, so its actually `expr` and not `EXPR`

**Comment by dantleech** [comment]
So you can update your content repository sucessfully?

Is the bug that it should be case insensative? (in addition to actually locating the exception class)

**Comment by dbu** [comment]
so the `SET EXPR()` syntax seems *not* case sensitive, but the `a.field = expr()` case only works lowercase. maybe that is by design - not sure but maybe functions should generally be case insensitive. would be more in line with the sql keywords being insenstitive.

`a.field = expr()` worked for me with beta3. `SET EXPR()` does not.

**Comment by dantleech** [comment]
Hmm, well `SET EXPR` is invalid as SET is always `field => value`. We could implement `APPLY EXPR()`

**Comment by dbu** [comment]
oh, but then the parser is really not reporting that one well. it executed something and the error comes on the auto-save. i will adjust the ticket description then
